src/librustc_span/lib.rs

```diff
@@ -1075,7 +1116,11 @@ impl SourceFile {
         unmapped_path: FileName,
         mut src: String,
         start_pos: BytePos,
+        checksum_kind: Option<SourceFileChecksumKind>,
     ) -> Self {
+        // Compute the file checksum *before* any normalization.
+        let src_file_checksum =
+            checksum_kind.and_then(|k| Some(SourceFileChecksum::new(k, &src.as_bytes())));
```

Comment:
Would it make more sense to use `?` instead of `and_then`?

Reply:
Maybe you meant `map`? I don't think I can use the `?` operator here.